services/registry/handlers/users.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const { response, fork } = require('boltzmann');
+const authn = require('../decorators/authn');
+
+module.exports = [
+  fork.get('/v1/users/user/:username/memberships', authn.required(memberships)),
+  fork.post('/v1/users/user/:username/memberships/invitation/:namespace@:host', authn.required(accept)),
+  fork.delete('/v1/users/user/:username/memberships/invitation/:namespace@:host', authn.required(decline))
+];
+
+async function memberships(context, { username }) {
+  const [err, result] = await context.storageApi.listUserMemberships({
+    for: username,
+    bearer: context.user.name,
+    page: context.url.query.page,
+    status: context.url.query.status
+  })
+
+  if (err) {
+    // TODO: enumerate error cases
+    return response.error('Caught error listing memberships', 500)
+  }
+
+  const { objects, next, prev, total } = result
+  return response.json({ objects, next, prev, total });
+}
+
+async function accept(context, { username, namespace, host }) {
+  const [err] = await context.storageApi.acceptNamespaceMembership({
+    bearer: context.user.name,
+    invitee: username,
+    namespace,
+    host
+  }).then(
+    xs => [null, xs],
```

Comment:
Would it be worth pulling this into an `either` util?

Reply:
That would be great once this lands!